doc/cephfs/fs-nfs-exports.rst

```diff
@@ -0,0 +1,58 @@
+=======================
+CephFS Exports over NFS
+=======================
+
+CephFS namespaces can be exported over NFS protocol using the
+`NFS-Ganesha NFS server <https://github.com/nfs-ganesha/nfs-ganesha/wiki>`_.
+
+Requirements
+============
+
+-  Latest Ceph file system with mgr and dashboard enabled
+-  'nfs-ganesha', 'nfs-ganesha-ceph' and nfs-ganesha-rados-grace packages
+   (version 2.7.6-2 and above)
+
+Create NFS Ganesha Cluster
+==========================
+
+.. code:: bash
+
+    $ ceph fs nfs cluster create [--size=1] <cluster_id>
+
+This creates a common recovery pool for all Ganesha daemons, new user based on
+cluster_id and common ganesha config rados object.
+
+.. note:: This does not setup ganesha recovery database and start the daemons.
+          It needs to be done manually if not using vstart for creating
+          clusters. Please refer `ganesha-rados-grace doc
+          <https://github.com/nfs-ganesha/nfs-ganesha/blob/next/src/doc/man/ganesha-rados-grace.rst>`_
+
+Create NFS Export
+==================
+
+.. code:: bash
+
+    $ ceph fs nfs export create --fs_name <fs_name> [--read-only] [--path=/path/in/cephfs] [--attach=<cluster_id>] <binding>
```

Comment:
nit: how about --fs-name or maybe --fsname instead of --fs_name? --read-only is all hyphens.

Extending what @sebastian-philipp suggested above, this should probably be something like:

`ceph nfs export create --type fs --fsname <fs_name> [--read-only] [--path=/path/in/cephfs] [--attach=<cluster_id>] <binding>`

...then in the future, someone could add a `--type rgw` create.

Reply:
It is good idea. Maybe later both clustering and export part will have to move out of volumes.
As it is now under volumes module, is it fine if I just add a todo note for it ?

Reply:
Sure. I assume we're targeting this feature for Pacific? We probably won't want to backport this to Octopus, or otherwise advertise the feature until the command-line syntax is settled.

Reply:
I would like all of the NFS work backported to octopus once it's ready.  We can wait until it's working in master first, though, instead of backporting bits and pieces as we go.

Reply:
Yes, fine with backporting it eventually, just want to avoid having two different variants of the command-line syntax in released versions. That's just confusing for users.

I think promoting this to a first class "ceph" command makes more sense long term as it's sort of its own thing and not really part of the fs. I'd prefer that syntax over "ceph fs nfs...".